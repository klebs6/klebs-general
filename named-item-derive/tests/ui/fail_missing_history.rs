// ---------------- [ File: tests/ui/fail_missing_history.rs ]
// Should fail because `history="true"` but no name_history field is present

use named_item_derive::NamedItem;

#[derive(NamedItem)]
#[named_item(history="true")]
struct MissingHistoryField {
    name: String,
}

fn main() {}
