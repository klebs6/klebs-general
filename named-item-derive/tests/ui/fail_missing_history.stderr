error: history=true but no `name_history: Vec<String>` field found.
 --> tests/ui/fail_missing_history.rs:8:8
  |
8 | struct MissingHistoryField {
  |        ^^^^^^^^^^^^^^^^^^^
